### DISCUSSION TITLE
ESP bei Neustart nicht verfügbar

### DISCUSSION THREAD

**Discussion by Tom-Bom-badil** [Tom-Bom-badil/samson_trovis_557x#60] [General (de/en)]
_Originally posted as issue by @BOFHELL in https://github.com/Tom-Bom-badil/samson_trovis_557x/issues/3#issuecomment-1298158510_

Ich nutze Modbus(RTU) in FHEM, zum Lesen/Schreiben **OHNE** Probleme

1. Wemo d1 Mini mit esp-link v3.2.47-g9c6530d
2. 5573-001  V.2.61, HW 129, Sonderfunktion 89
3. Anbindung:
- seitliche Buchse,
- Belegung wie im Wiki
- Versorgungsspannung komplett via Trovis*
4. Nutzung
- FHEM (Modbus RTU)
- Nutzung via mbusd und socat auf Linux für TrovisViewer mit esp-link möglich (Port 23)    
 `/usr/bin/socat -s -ly pty,link=/dev/trovis,setuid=root,raw tcp:IP.VOM.WEMO.D1:23`
 mbusd und esp-link stehen auf `19200 bauds` und `trx_control: rts`

*Für Firmware-Update funktionierte dies leider nicht, bin dafür auf den USRIOT USR-TCP232-T2 ausgewichen. 
Beim schreiben erhalte ich ab und an vom TrovisViewer zum Schluss eine Fehlermeldung "Connection Interrupted..." 
Alle Werte werden zwar geschrieben, denke es liegt am Timeout/Delay da die(meine) Trovis nach dem beschreiben einen Neustart durchführt und somit der Wemo keinen Strom mehr hat, was dann zum "Programmfehler" führt.

**Comment by Tom-Bom-badil** [comment]
Das könnte evtl an den Boot-Zeiten des ESP liegen, deshalb funktioniert der T2 da vermutlich besser.

Der Regler booted ziemlich fix, und wenn der ESP dann aufgrund Hard-Reset mit seiner Firmware noch nicht wieder 'richtig oben' ist, geht die PC-Verbindung vermutlich in's Leere. Interessant wäre hier mal, wann an dem +5V-Pin überhaupt wieder Spannung anliegt (vermutlich erst nach erfolgreichem Regler-Restart).

/tom

**Comment by BOFHELL** [comment]
Ja gehe ich von aus. der Wemo startet auch schnell, bis der dann wlan verbunden hat etc sagt software sicher timeout. oder mbusd. Ich wollte ohnehin den Kamstrup auslesen die Tage. Dann bekommt der Wemo ohnehin eigene Spannungsversorgung. Werd das dann korrigieren (kann aber auch paar Monate dauern)

**Comment by Tom-Bom-badil** [comment]
Der auf dem ESP verbaute 5V/3V3 Längsregler frisst eh viel Energie, gerade beim Booten.

Bei eigener Spannungsversorgung denk dran: Nur eine Spannungsversorgung sollte verbunden sein, sonst stirbt im ungünstigsten Fall DC/DC-Kette oder Längsregler.

/tom
